### DISCUSSION TITLE
Task - Exception Handling - String to Integer

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/python#15] [Tasks]
# Exceptions - String to Integer
#### Objective

Today, we're getting started with Exceptions by learning how to parse an integer from a string and print a custom error message.

#### Task

Read a string, `S` , and print its integer value; if `S` cannot be converted to an integer, print `Bad String`.

Note: You must use the String-to-Integer and exception handling constructs built into your submission language. If you attempt to use loops/conditional statements, you will get a `0` score.

#### Input Format

A single string, `S` .

#### Constraints

* 1 <= |S| <= 6 where |S| is the length of string `S`
* `S` is composed of either lowercase letters `(a-z)` or decimal digits `(0-9)`

#### Output Format

Print the parsed integer value of `S`, or `Bad String` if `S` cannot be converted to an integer.

#### Sample Input 0

```
3
```

#### Sample Output 0

```
2
```

#### Sample Input 1

```
za
```

#### Sample Output 1

```
Bad String
```

#### Explanation

Sample Case `0` contains an integer, so it should not raise an exception when we attempt to convert it to an integer. Thus, we print the `3`.

Sample Case `1` does not contain any integers, so an attempt to convert it to an integer will raise an exception. Thus, our exception handler prints `Bad String`.

## Given code

```python
import sys

S = input().strip()
```

**Comment by zenberr** [comment]
```python
try:
    print(int(input("Enter a string")))
except:
    print("Bad string")
```

**Comment by Edson-Rodrigo** [comment]
```python
string = input(" Enter a word or number: ").strip()

try:
    number = int(string)
    print(number)
except ValueError:
    print('Bad String')
```

**Comment by JesusBel16** [comment]
```python
inp = input()
try:
    inp = int(inp)
except:
    print("Bad string")
else:
    print(inp)
```

**Comment by AlfredoITS** [comment]
```python
import sys

S = input("Enter a number or a word: ").strip()

try:
    print("The number is:", int(S))
except ValueError: 
    print(S,"is Bad String.")
```

**Comment by KevinIturbe** [comment]
## String to Integer

````python
s = input().strip()

try:
    print(int(s))
except:
    print("Bad string")
````

**Comment by OlvarM** [comment]
# String to Int
``` python
# string to int using exception handling
def main():
    try:
        print(int(input("Enter a number: ")))
    except:
        print("Bad Number")

if __name__ == "__main__":
    main()
```

**Comment by ElijahHorowitz** [comment]
```python
import sys

s = input().strip()
try:
    print(int(s))
except:
    print("Bad String")
```

**Comment by rama77605** [comment]
```python
string = input(" Enter a word or number: ").strip()

try:
    num = int(string)
    print(num)
except ValueError:
    print('Bad String')
```

**Comment by Jyothig7** [comment]
1.
```python
import sys

String = input("Enter a number or word : ").strip()

try:
    print("The number is :", int(String))
except ValueError: 
    print(S," Bad String.")
```

**Comment by eusoj2** [comment]
1.-
```Python
S = input().strip()
try:
    int(S)
    print(S)
except:
    print("Bad String")
```
